src/components/AnchorList/index.js

```diff
@@ -0,0 +1,179 @@
+import React, { useState } from 'react'
+import PropTypes from 'prop-types'
+import Modal from '../Modal'
+import {
+  AncherContainer,
+  AnchorContent,
+  AnchorTitle,
+  AnchorLinksMob,
+  AnchorLinks,
+  AnchorIconLink,
+  AnchorIconTitle,
+  AnchorContainerDes,
+  AnchorLinksContainerDes,
+  AnchorContentDes,
+  AnchorItemsDes,
+  AnchorItemsText,
+} from './style'
+import Icon from '~/components/Icon'
+import { WidthLimiterContainer } from '../Layout/styles'
+import { Tablet, Desktop } from '../Utilities/Media'
+import Text from '../Utilities/Text'
+import Img from 'gatsby-image'
+
+const AnchorList = ({ anchorQuery }) => {
+  const [open, setOpen] = useState(false)
+  const [ministerState, setMinisterState] = useState(0)
+  const [stateQueury, setStateQuery] = useState('')
+
+  const links = []
+
+  anchorQuery?.links
+    ? anchorQuery.links.map((info) => {
+        Object.values(info).map((values) => {
+          links.push(values)
+        })
+      })
+    : ''
+
+  const handelClickMobile = (e) => {
+    setOpen(true)
+    setMinisterState(links.indexOf(e.target.innerText))
+    setStateQuery(e.target.innerText)
+  }
+
+  const handelClick = (e) => {
+    setStateQuery(e.target.innerText)
+    setMinisterState(links.indexOf(e.target.innerText))
+  }
+
+  const anchorLinksMobile = links.map((link, index) => {
+    return (
+      <AnchorLinksMob key={index}>
+        <AnchorIconLink>
+          <Icon type="add" />
+        </AnchorIconLink>
+        <div className="anchor_links" onClick={handelClickMobile}>
+          {link ? link : ''}
+        </div>
+      </AnchorLinksMob>
+    )
+  })
+
+  const anchorArray = []
+
+  const anchorFirstImage = anchorQuery?.ministries[0]?.list_image?.localFile
+    ?.childImageSharp?.fluid
+    ? anchorQuery.ministries[0].list_image.localFile.childImageSharp.fluid
+    : ''
+
+  const anchorFirstText = anchorQuery?.ministries[0]?.list_text
+    ? anchorQuery.ministries[0].list_text
+    : ''
+
+  anchorQuery?.ministries
+    ? anchorQuery.ministries.map((info) => {
+        if (info.list_title.text === stateQueury) {
+          anchorArray.push(info)
+        }
+      })
+    : ''
+
+  const anchorLinks = links.map((link, index) => {
+    return (
+      <AnchorLinks key={index}>
+        <AnchorIconLink>
+          <Icon type="add-des" />
+        </AnchorIconLink>
+        <div
+          className="anchor_links"
+          style={
+            ministerState === index ? { color: '#CC1D27' } : { color: 'black' }
+          }
+          onClick={handelClick}
+        >
+          {link ? link : ''}
+        </div>
+      </AnchorLinks>
+    )
+  })
+
+  return (
+    <AncherContainer>
+      <WidthLimiterContainer>
+        <AnchorContent>
+          <Tablet>
+            <AnchorTitle as="h2" type="heading2">
+              <AnchorIconTitle>
+                {anchorQuery?.title ? anchorQuery.title : ''}
```

Comment:
Safety check outside the AnchorIconTitle (Reason being that if your title is empty and the AnchorIconTitle has css properties like margin-bottom, it will still show and break layout)